src/ast.rs

```diff
@@ -446,6 +446,16 @@
     }
 }
 
+/// This is for optional values and for lists.
+trait CheckParse {
+    /// A true return means we should descend into the production, false means stop.
+    fn can_be_parsed(pop: &mut Populator<'_>) -> bool;
+}
+
+trait StaticTokenInfo {
```

Comment:
is it usual to prefix type-only traits with `Static`?

Reply:
It is not, but I wasn't sure how else to name it. I wanted to put the item in the regular `Token` trait, but object safety got in the way.
We're not actually using it abstractly, so I'll move it to an inherent `impl`.